src/flag_gems/ops/angle.py

```diff
@@ -0,0 +1,39 @@
+import math
+
+import torch
+import triton
+import triton.language as tl
+
+from ..utils import pointwise_dynamic, tl_extra_shim
+
+atan2 = tl_extra_shim.atan2
+
+
+@pointwise_dynamic(is_tensor=[True, True], promotion_methods=[(0, "DEFAULT")])
+@triton.jit
+def angle_func(real, imag):
+    real_last = tl.where(real.dtype == torch.float16, real.to(tl.float32), real)
+    imag_last = tl.where(imag.dtype == torch.float16, imag.to(tl.float32), imag)
```

Comment:
just upcast them and tl.where is unnecessary.

Reply:
Or use if else expression?`
I wonder whether this would translate into a `SELECT` instruction, since `real.dtype == torch.float16` is a constexpr condition.